### PR TITLE
Add Gocritic (1/2)

### PR DESCRIPTION
**Pull request by marbar3778**
- Add `gocritic` as a linter

- The linting is not complete, but should i complete in this PR or in a following. 
- 23 files have been touched so it may be better to do in a following PR

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: consensus/mempool_test.go [hunk 1]
```diff
@@ -82,14 +82,14 @@ func TestMempoolProgressInHigherRound(t *testing.T) {
 	ensureNewRound(newRoundCh, height, round) // first round at first height
 	ensureNewEventOnChannel(newBlockCh)       // first block gets committed
 
-	height = height + 1 // moving to the next height
+	height += 1 // moving to the next height
```
Comment (melekes) [inline consensus/mempool_test.go]:
```suggestion
	height++ // moving to the next height
```

File: consensus/mempool_test.go [hunk 2]
```diff
@@ -82,14 +82,14 @@ func TestMempoolProgressInHigherRound(t *testing.T) {
...
 	round = 0
 
 	ensureNewRound(newRoundCh, height, round) // first round at next height
 	deliverTxsRange(cs, 0, 1)                 // we deliver txs, but dont set a proposal so we get the next round
 	ensureNewTimeout(timeoutCh, height, round, cs.config.TimeoutPropose.Nanoseconds())
 
-	round = round + 1                         // moving to the next round
+	round += 1                                // moving to the next round
```
Comment (melekes) [inline consensus/mempool_test.go]:
```suggestion
	round++                                // moving to the next round
```

File: consensus/state_test.go [hunk 3]
```diff
@@ -181,7 +181,7 @@ func TestStateBadProposal(t *testing.T) {
 	propBlock, _ := cs1.createProposalBlock() //changeProposer(t, cs1, vs2)
 
 	// make the second validator the proposer by incrementing round
-	round = round + 1
+	round += 1
```
Comment (melekes) [inline consensus/state_test.go]: ++

File: consensus/state_test.go [hunk 4]
```diff
@@ -374,7 +374,7 @@ func TestStateLockNoPOL(t *testing.T) {
 
 	///
 
-	round = round + 1 // moving to the next round
+	round += 1 // moving to the next round
```
Comment (melekes) [inline consensus/state_test.go]: ++

File: consensus/state_test.go [hunk 5]
```diff
@@ -418,7 +418,7 @@ func TestStateLockNoPOL(t *testing.T) {
 	// then we enterPrecommitWait and timeout into NewRound
 	ensureNewTimeout(timeoutWaitCh, height, round, cs1.config.Precommit(round).Nanoseconds())
 
-	round = round + 1 // entering new round
+	round += 1 // entering new round
```
Comment (melekes) [inline consensus/state_test.go]: you get the idea

File: crypto/secp256k1/internal/secp256k1/curve.go [hunk 6]
```diff
@@ -86,83 +86,83 @@ type BitCurve struct {
 	BitSize int      // the size of the underlying field
 }
 
-func (BitCurve *BitCurve) Params() *elliptic.CurveParams {
+func (bitCurve *BitCurve) Params() *elliptic.CurveParams {
```
Comment (melekes) [inline crypto/secp256k1/internal/secp256k1/curve.go]: better to leave the fork untouched

Comment (marbar3778) [reply crypto/secp256k1/internal/secp256k1/curve.go]: 👍 will revert

**Comment by marbar3778** [comment]
seems test_abci_cli is getting hung up on the server, locally not able to recreate the issue

**Pull request merged**